src/pages/studies/Study.tsx

```diff
@@ -91,31 +91,32 @@ const Study: React.FC = () => {
     }
   };
 
-  const handleDownloadAuthKey = async () => {
+  const handleDownloadFile = async (url: string, fileName: string) => {
     try {
-      const response = await fetch(`${apiBaseUrl}/api/download_auth_key?study_id=${study_id}`, {
-        method: "GET",
-        headers,
-      });
-
-      if (!response.ok) {
-        throw new Error((await response.json()).error || "Unexpected error");
+      const res = await fetch(url, { headers });
+      if (!res.ok) {
+        throw new Error(await res.text());
       }
-
-      const data = await response.blob();
-      const url = window.URL.createObjectURL(data);
-      const a = document.createElement("a");
-      a.style.display = "none";
-      a.href = url;
-      a.download = "auth_key.txt";
+      const blob = await res.blob();
+      const objUrl = URL.createObjectURL(blob);
+      const a = document.createElement('a');
+      a.href = objUrl;
+      a.download = fileName;
       document.body.appendChild(a);
       a.click();
-      window.URL.revokeObjectURL(url);
-    } catch (error) {
-      console.error("Failed to download auth key:", error);
+      document.body.removeChild(a);
+      URL.revokeObjectURL(objUrl);
+    } catch (err) {
+      console.error(`Failed to download ${fileName}:`, err);
     }
   };
 
+  const handleDownloadSAKey = () =>
```

Comment:
Is this stored in local or session storage? And can instead this be handled by a token and refresh token? 
Session storage is more brittle, but safer for long lived secrets.

Reply:
Luckily, not store anywhere in the browser, and is just requested and downloaded on demand by the user